src/ray/raylet/scheduling/cluster_resource_scheduler.h

```diff
@@ -83,16 +85,22 @@ class ClusterResourceScheduler {
   ///  prioritize_local_node if set to true.
   ///  \param requires_object_store_memory: take object store memory usage as part of
   ///  scheduling decision.
-  ///  \param is_infeasible[out]: It is set true if the task is not schedulable because it
-  ///  is infeasible.
+  ///  \param is_infeasible[out]: It is set
+  ///  true if the task is not schedulable because it is infeasible.
+  ///  \param forward_to: The node to be forwarded to for scheduling.
+  ///  \param use_required_resources: Whether using required_resources (instead of
+  ///  required_placement_resources) for scheduling.
   ///
   ///  \return emptry string, if no node can schedule the current request; otherwise,
   ///          return the string name of a node that can schedule the resource request.
-  scheduling::NodeID GetBestSchedulableNode(const TaskSpecification &task_spec,
-                                            bool prioritize_local_node,
-                                            bool exclude_local_node,
-                                            bool requires_object_store_memory,
-                                            bool *is_infeasible);
+  scheduling::NodeID GetBestSchedulableNode(
+      const TaskSpecification &task_spec,
+      bool prioritize_local_node,
+      bool exclude_local_node,
+      bool requires_object_store_memory,
+      bool *is_infeasible,
+      scheduling::NodeID forward_to = scheduling::NodeID::Nil(),
```

Comment:
this seems wrong. In this case I think we should directly talk to remote cluster_task_manager, instead of adding this forward_to logic to cluster task manager. And the logic should ideally lives in gcs_actor_scheduler.  Is there any reason we can't do that?

Reply:
If there is not any node in the cluster, then actor has to be pending in gcs, waiting for resource update (no matter which scheduler is used). Previously, the pending queue is within gcs_actor_manager. Now we want all queueing handled by cluster_task_manager. So when dequeueing from cluster_task_manager, we use forward_to to decide whether we have to go through a scheduling_policy. @scv119

Reply:
I think I agree with @scv119 here: we must have at least one alive node in the cluster right? The queueing should be handled by the remote cluster_task_manager?

Reply:
> I think I agree with @scv119 here: we must have at least one alive node in the cluster right? 
I see. So now my understanding is: when gcs receives an actor creation request, it means there must be a driver/worker running (which sent this request) out there. And the pre-condition that a worker can be running is that a local raylet exists. So the scheduler in gcs can definitely assume there is at least one alive node which can handle the queueing. Is this right? @jjyao @scv119